README.md

```diff
@@ -51,7 +58,233 @@ custom features to work with multiple nodes and retry logic.
 
-# Transaction Statuses returned by the library
+## ClientBuilder Methods
+
+Client allows you to create your own client with setting it with multiple nodes and custom http client.
+
+### WithArc Method
+
+```go
+// Set the access token and url for the node
+ token := ""
+ apiURL := "https://tapi.taal.com/arc"
+
+ cfg := broadcast_client.ArcClientConfig{
+  Token:  token,
+  APIUrl: apiURL,
+ }
+
+ client := broadcast_client.Builder().
+  WithArc(cfg).
+  Build()
+```
+
+We can also call multiple times the method `WithArc` to set multiple nodes.
+
+```go
+// Set the access token and url for the node
+ client := broadcast_client.Builder().
+  WithArc(cfg1).
+  WithArc(cfg2).
+  WithArc(cfg3).
+  Build()
+```
+
+What is the call order if we have multiple nodes configured?
+
+We use the **strategy** to define the order of the calls. The default strategy is `OneByOne` in RoundRobin algorith that will call the nodes in the order they were set.
+
+### WithHTTPClient Method
+
+```go
+// (...)
+ clent := broadcast_client.Builder().
+  WithArc(cfg).
+  WithHTTPClient(&http.Client{}).
+  Build()
+```
+
+We can use the method `WithHTTPClient` to set the custom http client.
+It needs to implement the interface `HTTPInterface` that is defined in the [httpclient.go](/broadcast/internal/httpclient/http_client.go#L35)
+
+```go
+type HTTPInterface interface {
+ DoRequest(ctx context.Context, pld HTTPRequest) (*http.Response, error)
+}
+```
+
+**Only if all of them fail, we will return the error to the user.**
+
+## ARC Client Methods
+
+### QueryTx Method
+
+When you created your own client, you can use the method `QueryTx` to query the status of a transaction.
+
+```go
+    // ...
+    hex := "9c5f5244ee45e8c3213521c1d1d5df265d6c74fb108961a876917073d65fef14"
+
+    result, err := client.QueryTransaction(context.Background(), hex)
+    // ...
+```
+
+```go
+func (a *ArcClient) QueryTransaction(ctx context.Context, txID string) (*broadcast.QueryTxResponse, error) {
+ if a == nil {
+  return nil, broadcast.ErrClientUndefined
+ }
+
+ result, err := queryTransaction(ctx, a, txID)
+ if err != nil {
+  return nil, err
+ }
+
+ err = validateQueryTxResponse(result)
+ if err != nil {
+  return nil, err
+ }
+
+ return result, nil
+}
+```
+
+1. The method `QueryTransaction` is calling the function `queryTransaction` that is trying to query the status of the transaction from the node. It uses the HTTP GET protocol to do so.
+
+2. Then we validate the response from the node using the function `validateQueryTxResponse`. This function is checking if the response is valid and if the status code is 200.
+
+3. If everything is ok, we return the response to the user.
+
+### SubmitTx Method
+
+Having your client created, you can use the method `SubmitTx` to submit a single transaction to the node.
+
+```go
+    // ...
+    tx := broadcast.Transaction{
+      RawTx: "xyz",
+    }
+
+    result, err := client.SubmitTransaction(context.Background(), tx)
+    // ...
+```
+
+```go
```

Comment:
I'm not sure we need/want this kind of code samples and information about HTTP verbs used. 
As I understand our broadcast client is an abstraction over Arc Rest API and it should hides implementation details from clients